rskj-core/src/main/java/co/rsk/peg/whitelist/LockWhitelist.java

```diff
@@ -64,7 +64,12 @@ public boolean isWhitelisted(byte[] address) {
 
     public boolean isWhitelistedFor(Address address, Coin amount, int height) {
         Coin maxTransferValue = getMaxTransferValue(address);
-        return height > disableBlockHeight || (isWhitelisted(address) && maxTransferValue !=null && (amount.isLessThan(maxTransferValue) || amount.equals(maxTransferValue)));
+        if (height > disableBlockHeight) {
+            // Whitelist disabled
+            return true;
+        }
+
+        return (isWhitelisted(address) && (amount.isLessThan(maxTransferValue) || amount.equals(maxTransferValue)));
```

Comment:
I believe this method should leave its answer to the lock whitelist entry itself.